task/sql-tasks.js

```diff
@@ -280,7 +460,26 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT DISTINCT
+        Customers.CompanyName,
+        Products.ProductName,
+        OrderDetails.UnitPrice as "PricePerItem"
+    FROM Customers
+    INNER JOIN Orders ON Orders.CustomerID = Customers.CustomerID
+    INNER JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+    INNER JOIN Products ON OrderDetails.ProductID = Products.ProductID
+    WHERE OrderDetails.UnitPrice = (
+        SELECT 
+            MAX(OrdDet.UnitPrice)
+        FROM Customers as Cust
+        INNER JOIN Orders AS Ord ON Cust.CustomerID = Ord.CustomerID 
+        INNER JOIN OrderDetails AS OrdDet ON Ord.OrderID = OrdDet.OrderID
+        WHERE Cust.CompanyName = Customers.CompanyName

```

Comment:
company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.